libraries/codegen-tooling/rules.mk

```diff
@@ -0,0 +1,11 @@
+# Defines $(T)_SRC, $(T)_INC, $(T)_DEPS, and $(T)_CFLAGS for the build makefile.
```

Comment:
I agree on having a separate directory/library for autogenerated files. I think it's useful to make it explicit what files are autogenerated and what files aren't.

Ideally autogenerated files should contain `autogenerated` in their filename, or at least contain a comment saying that they are autogenerated is probably a good idea. At the very least, add a comment in this `rules.mk` file saying that the files here are the autogenerated output.